### PR TITLE
Update to Ubuntu 24.04

### PR DESCRIPTION
**Pull request by sligocki**
Hey Matt, what do you think about updating this to be based on a new version of Ubuntu? It looks like this was pinned to Ubuntu 20.04 (from 2020). This includes python 3.8 which is starting to get pretty different than the current 3.12 (as you and Myles noted when my type annotations started causing parsing errors)?

I'm sure we'll need to do some testing once we switch to make sure the new version works well, so no need to move quickly on merging this PR, but just wanted to get the ball rolling.

### CODE REVIEW

**Comment by sligocki** [comment]
Oops, looks like I need to get this building first :)

**Comment by mstrimas** [comment]
Totally agree, and actually tried to do that yesterday, but then rolled back to address that disk space issue first. Feel free to rip out that `RUN useradd docker` bit that's causing the error, I think I was just copying some stuff that was from the rocker R images.

**Comment by mstrimas** [comment]
@sligocki trying awcli install instructions from https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html

**Comment by mstrimas** [comment]
...aaaand now we're stuck on some Python package install error that I'm sure you're more qualified than me to troubleshoot

**Comment by sligocki** [comment]
Yeah, I'm working on this.

**Comment by mstrimas** [comment]
It appears like this succeeded, but `stringr` wasn't installed, which prevented a bunch of other R packages:
```
#12 7.809  .onAttach failed in attachNamespace() for 'ebirdst', details:
#12 7.809   call: dyn.load(file, DLLpath = DLLpath, ...)
#12 7.809   error: unable to load shared object '/opt/R/4.4.0/lib/R/library/stringi/libs/stringi.so':
#12 7.809   libicui18n.so.66: cannot open shared object file: No such file or directory
```

**Comment by mstrimas** [comment]
I was able to resolve this locally by install the development version of `stringi` from GitHub. Trying to see if that works with GitHub actions build now.

File: geospatial/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,10 @@
 # add r and python geospatial tools
 FROM mstrimas/tidyverse
 
-MAINTAINER "Matt Strimas-Mackey" mes335@cornell.edu
 LABEL maintainer="Matt Strimas-Mackey <mes335@cornell.edu>"
 
 # install spatial packages, from https://hub.docker.com/r/rocker/geospatial/dockerfile
 RUN apt-get update \
-  # connect to ubuntugis/ubuntugis-unstable to get latest versions of spatial libraries
-  && add-apt-repository ppa:ubuntugis/ubuntugis-unstable -y \
```
Comment (sligocki) [inline geospatial/Dockerfile:10]: @mstrimas what do you think about removing this? It looks like it hasn't been ported to Noble yet (which I guess is only a month old, lol). I'm guessing Noble might just have pretty recent versions of these GIS packages to begin with, but I figured you'd know some of the details of why we needed this originally.

**Comment by mstrimas** [comment]
@sligocki I'm testing this now on Bridges2 and it appears we `pyarrow` has been dropped

**Comment by mstrimas** [comment]
Also, there's some issue with the `arrow` R package:

```
> library(arrow)
Error: package or namespace load failed for ‘arrow’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/opt/R/4.4.0/lib/R/library/arrow/libs/arrow.so':
  libcrypto.so.1.1: cannot open shared object file: No such file or directory
```

**Comment by mstrimas** [comment]
> Also, there's some issue with the `arrow` R package:
> 
> ```
> > library(arrow)
> Error: package or namespace load failed for ‘arrow’ in dyn.load(file, DLLpath = DLLpath, ...):
>  unable to load shared object '/opt/R/4.4.0/lib/R/library/arrow/libs/arrow.so':
>   libcrypto.so.1.1: cannot open shared object file: No such file or directory
> ```

just pushed a change that should install this properly, leaving the `pyarrow` fix to you

**Comment by sligocki** [comment]
Lol apparently python3-arrow is a totally different package than pyarrow!

**Comment by mstrimas** [comment]
Looks like we've hit some sort of rate limit for the day for the GH actions. I also discovered a couple more R packages that aren't working while testing on Bridges2. I'll wait a while for the rate limit to reset, the push with the R package fixes.

**Comment by mstrimas** [comment]
Two more R package install errors that are again related to not finding some shared library:

```
Error: package or namespace load failed for ‘mapboxapi’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/opt/R/4.4.0/lib/R/library/magick/libs/magick.so':
  libMagick++-6.Q16.so.8: cannot open shared object file: No such file or directory
```
and
```
#12 30.01 Error in dyn.load(file, DLLpath = DLLpath, ...) : 
#12 30.01   unable to load shared object '/opt/R/4.4.0/lib/R/library/protolite/libs/protolite.so':
#12 30.01   libprotobuf.so.17: cannot open shared object file: No such file or directory
```

Annoying the GitHub Actions shows a successful build despite these errors. The only way I know to fix these is to compile the problem packages from source, but if you happen to know a better way to fix these shared library problems, let me know!

**Comment by mstrimas** [comment]
I just finished testing this through all the stages of the production workflow. Looks great to me, merging now!

**Pull request merged**